kfs-core/src/main/java/edu/arizona/kfs/sys/document/validation/impl/AccountingLineGlobalTransactionValidation.java

```diff
@@ -0,0 +1,166 @@
+     * @return true if object code is allowed, given the account's sub fund
+     */
+    private boolean validateAccountingLine(AccountingDocument financialDocument, AccountingLine accountingLine) {
+        LOG.debug("beginning object code validation ");
+
+        boolean lineAllowed = true;
+        if (isAccoutingLineComplete(accountingLine)) {
+            Message msg = globalTransactionEditService.isAccountingLineAllowable((AccountingLineBase) accountingLine, financialDocument.getFinancialDocumentTypeCode());
+            if (msg != null) {
+                GlobalVariables.getMessageMap().putError(KFSConstants.AMOUNT_PROPERTY_NAME, KFSKeyConstants.ERROR_CUSTOM, msg.getMessage());
+                lineAllowed = false;
+            }
+        }
+
+        return lineAllowed;
+    }
+
+    /**
+     * Tests the accounting line to see if there are sufficient values in it to validate.
+     * <p>
+     * For example, there needs to be at least a chartOfAccountsCode, an accountNumber, and
+     * an objectCode, otherwise the validation will fail.
+     *
+     * @param accountingLine
+     * @return
+     */
+    private boolean isAccoutingLineComplete(AccountingLine accountingLine) {
+        accountingLine.refreshReferenceObject("account");
+        if (ObjectUtils.isNull(accountingLine.getAccount())) {
+            return false;
+        }
+
+        accountingLine.getAccount().refreshReferenceObject("subFundGroup");
+        if (ObjectUtils.isNull(accountingLine.getAccount().getSubFundGroup())) {
+            return false;
+        }
+
+        accountingLine.refreshReferenceObject("objectCode");
+        if (ObjectUtils.isNull(accountingLine.getObjectCode())) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Gets the accountingDocumentForValidation attribute.
+     *
+     * @return Returns the accountingDocumentForValidation.
+     */
+    public AccountingDocument getAccountingDocumentForValidation() {
+        return accountingDocumentForValidation;
+    }
+
+    /**
+     * Sets the accountingDocumentForValidation attribute value.
+     *
+     * @param accountingDocumentForValidation The accountingDocumentForValidation to set.
+     */
+    public void setAccountingDocumentForValidation(AccountingDocument accountingDocumentForValidation) {
+        this.accountingDocumentForValidation = accountingDocumentForValidation;
+    }
+
+    /**
+     * Sets the accountingLineForValidation attribute value.
+     *
+     * @param accountingLineForValidation The accountingLineForValidation to set.
+     */
+    public void setAccountingLineForValidation(AccountingLine accountingLineForValidation) {
+        this.accountingLineForValidation = accountingLineForValidation;
+    }
+
+    /**
+     * Sets the parameterService attribute value.
+     *
+     * @param parameterService The parameterService to set.
+     */
+    public void setParameterService(ParameterService parameterService) {

```

Comment:
I am not seeing parameter service used anywhere in this class. This class also has the old license.